### PR TITLE
Refactor/plugins

### PR DESCRIPTION
**Pull request by Tedc0819**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ module.exports = {
   Router: require('./lib/models/Router.js'),
   AppError: require('./lib/services/AppError.js'),
   HandlerRouter: require('./lib/services/HandlerRouter.js'),
-  Logger: require('./lib/services/Logger.js'),
-  MessageQueue: require('./lib/services/MessageQueue.js'),
+  Logger: require('./lib/plugins/logger/lib/Logger.js'),
   QueueTask: require('./lib/services/QueueTask.js'),
-  Redis: require('./lib/services/Redis.js'),
+  MessageQueue: require('./lib/plugins/messageQueue/lib/RabbitMessageQueue.js'),
```
Comment (Seitk) [inline index.js]: @Tedc0819 see if we want to change to expose classes within the plugin

File: lib/plugins/logger/lib/LoggerPlugin.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class LoggerPlugin {
+  didLoadFramework(app) {
+    const {
+      config,
+    } = app
+
+    app.logger = new Logger(config.logger)
+
+    Object.assign(
+      app.context,
```
Comment (Seitk) [inline lib/plugins/logger/lib/LoggerPlugin.js]: sounds bad to let the plugin freely override the app object lol

File: lib/plugins/logger/lib/LoggerPlugin.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      }
+    )
+
+  }
+
+  prepare(app) {
+
+  }
+
+  async connectDependencies(app) {
```
Comment (Seitk) [inline lib/plugins/logger/lib/LoggerPlugin.js]: maybe remove empty implementation of given phases

File: lib/plugins/logger/lib/LoggerPlugin.js (hunk 2 above)
Comment (hinxcode) [reply lib/plugins/logger/lib/LoggerPlugin.js]: seems not too bad if it override log and Logger only into context?

> sounds bad to let the plugin freely override the app object lol

seems not too bad if it override log and Logger only into context?

**Pull request merged**